doc/sphinx/practical-tools/utilities.rst

```diff
@@ -478,6 +485,46 @@ The build infrastructure generated by ``coq_makefile`` uses ``coqdep`` to
 automatically compute the dependencies among the files part of the
 project.
 
+Building a |Coq| project with Dune
+----------------------------------
+
+_note_: the canonical documentation for the Coq Dune extension is
+maintained upstream; please refer to the Dune manual for updated
+information.
+
+Building a Coq development with Dune requires setting up a Dune
+project for your files. This involves adding a ``dune-project`` and
+``pkg.opam`` file to the root, and providing ``dune`` files in the
+directories your ``.v`` files are placed:
+
+.. code:: scheme
+
+    (coqlib
+     (name <module_prefix>)
+     (public_name <package.lib_name>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (libraries <ocaml_libraries>)
+     (flags <coq_flags>))
+
+This stanza will build all `.v` files on the given directory, wrapping
+the library under ``<module_prefix>``. If you declare a
+``<package.lib_name>`` a ``.install`` file for the library will be
+generated; the optional ``<modules>`` field does allow you to filter
+the list of modules, and ``<libraries>`` allows to depend on ML plugins.
```

Comment:
Maybe note that in the first version of the Dune support, the Coq libraries will be found with the usual Coq support for this, including `$COQPATH`.